types/schemaoptions.d.ts

```diff
@@ -0,0 +1,183 @@
+     * This method accepts a few options. Instead of applying these options on a per-document basis, we may
+     * declare the options at the schema level and have them applied to all of the schema's documents by
+     * default.
+     */
+    toObject?: ToObjectOptions;
+    /**
+     * By default, if you have an object with key 'type' in your schema, mongoose will interpret it as a
+     * type declaration. However, for applications like geoJSON, the 'type' property is important. If you want to
+     * control which key mongoose uses to find type declarations, set the 'typeKey' schema option.
+     */
+    typeKey?: string;
+
+    /**
+     * By default, documents are automatically validated before they are saved to the database. This is to
+     * prevent saving an invalid document. If you want to handle validation manually, and be able to save
+     * objects which don't pass validation, you can set validateBeforeSave to false.
+     */
+    validateBeforeSave?: boolean;
+    /**
+     * The versionKey is a property set on each document when first created by Mongoose. This keys value
+     * contains the internal revision of the document. The versionKey option is a string that represents
+     * the path to use for versioning. The default is '__v'.
+     *
+     * @default '__v'
+     */
+    versionKey?: string | boolean;
+    /**
+     * By default, Mongoose will automatically select() any populated paths for you, unless you explicitly exclude them.
+     *
+     * @default true
+     */
+    selectPopulatedPaths?: boolean;
+    /**
+     * skipVersioning allows excluding paths from versioning (i.e., the internal revision will not be
+     * incremented even if these paths are updated). DO NOT do this unless you know what you're doing.
+     * For subdocuments, include this on the parent document using the fully qualified path.
+     */
+    skipVersioning?: {[key: string]: boolean; };
+    /**
+     * Validation errors in a single nested schema are reported
+     * both on the child and on the parent schema.
+     * Set storeSubdocValidationError to false on the child schema
+     * to make Mongoose only report the parent error.
+     */
+    storeSubdocValidationError?: boolean;
+    /**
+     * The timestamps option tells mongoose to assign createdAt and updatedAt fields to your schema. The type
+     * assigned is Date. By default, the names of the fields are createdAt and updatedAt. Customize the
+     * field names by setting timestamps.createdAt and timestamps.updatedAt.
+     */
+    timestamps?: boolean | SchemaTimestampsConfig;
+
+    /**
+     * Using `save`, `isNew`, and other Mongoose reserved names as schema path names now triggers a warning, not an error.
+     * You can suppress the warning by setting { supressReservedKeysWarning: true } schema options. Keep in mind that this
+     * can break plugins that rely on these reserved names.
+     */
+     supressReservedKeysWarning?: boolean
```

Comment:
Actually this is a typo. Should have been "suppress[...]" and not "supress[...]".

Reply:
You're right, that's something we should change in Mongoose 7. I opened up #11495 to track that